text/083-boost-stdlib-dev/text.md

```diff
@@ -0,0 +1,786 @@
+  * <details><summary>contains:</summary>
+    theories/derive/Derive.v</details>
+* coq-roots-rtauto
+  * The Coq standard library, rtauto
+  * Namespace `Coq.rtauto`
+  * 2 files, 0.8 kloc
+  * <details><summary>contains:</summary> rtauto/Bintree.v,
+    rtauto/Rtauto.v</details>
+* coq-roots-wellfounded
+  * The Coq standard library, wellfounded
+  * Namespace `Coq.wellfounded`
+  * 9 files, 0.9 kloc
+  * <details><summary>contains:</summary> Wellfounded/Inclusion.v,
+    Wellfounded/Wellfounded.v, Wellfounded/Union.v,
+    Wellfounded/Transitive_Closure.v, Wellfounded/Well_Ordering.v,
+    Wellfounded/Inverse_Image.v, Wellfounded/Disjoint_Union.v,
+    Wellfounded/Lexicographic_Product.v,
+    Wellfounded/Lexicographic_Exponentiation.v</details>
+* coq-roots
+  * The Coq standard library
+  * A metapackage for all coq-roots-* packages constituting
+    the historical Coq standard library, that used to be shipped with Coq.
+  * 508 files, 183.4 kloc
+
+theories/ssr and theories/ssrmatching should go back into mathcomp
+(either in coq-mathcomp-ssreflect or in a new coq-mathcomp-ssr
+containing only that).
+
+### Documentation
+
+Need to document the above split in packages (could help structure the doc).
+
+## Maintainers
+
+We should do an open call for maintainers. Current maintainers (stdlib
+team from Coq on github) will become roots maintainer, they can of
+course opt out at any time.
+
+# Drawbacks
+
+The stdlib will no longer come bundled with Coq. It will however be
+just as easy to install via the usual package managers. More notably
+it will remain an integral part of the Coq Platform.
+
+# Alternatives
+
+The technical changes proposed here can stand on their own merits, and can be adopted without ratifying the perspective presented for motivating them. Further, each of the three splits discussed under [Packages](https://github.com/proux01/ceps/blob/boost-stdlib-dev/text/083-boost-stdlib-dev/text.md#packages) could be performed without the other ones, with or without the broader vision that led to them.
+
+1. The logical structure of the library could be split into packages, enforced by the build system, without user-visible structural changes or renaming of stdlib.
+2. The standandard library could be moved to its own repository without changing its user-facing relationship with Coq and non-stdlib packaes.
+3. Specific parts of the standard library could be spun out to their own repositories without changing the rest.
+
+Out of these, 1. has the advantage that it retains most of the technical refactoring done for this CEP, and achieves some of the technical goals listed here such as allowing users to `Require` only a known subset, while minimizing disruption to users in the face of multiple possible futures for the standard library more generally. In particular, keeping the boundaries between subparts of the standard library internal to Coq as opposed to exposing them to users would simplify later changing them.
```

Comment:
Could you list them [some of the technical goals] precisely?

Reply:
The most important ones to me are

- Making it clear what files are fair game to Require in each part of stdlib
- Making it clear what will be brought in by a Require of some stdlib file
- While this CEP makes no stability promises for these boundaries, at least it gives names for the components so we can talk about how stable or not they are, and specifically whether we may want to deprecate one or another component

Reply:
Thanks, integrated to the CEP (feel free to further edit if I made something wrong)